src/openms/source/FORMAT/HANDLERS/MzMLHandler.cpp

```diff
@@ -3883,17 +3883,18 @@ namespace OpenMS::Internal
       //--------------------------------------------------------------------------------------------
       //
 
+      // precursor m/z may come from "isolation window":
+      mz = precursor.getMetaValue("selected ion m/z",
+                                  precursor.getMZ());
+
       if (options_.getForceTPPCompatability() ||
           precursor.getCharge() != 0 ||
           precursor.getIntensity() > 0.0 ||
           precursor.getDriftTime() >= 0.0 ||
           precursor.getDriftTimeUnit() == DriftTimeUnit::FAIMS_COMPENSATION_VOLTAGE ||
           !precursor.getPossibleChargeStates().empty() ||
-          precursor.getMZ() > 0.0)
+          mz > 0.0)
```

Comment:
can someone comment if this is the proper way? e.g., always write selected ion m/z to be consistent?